ExperimentRunner/WebExperiment.py

```diff
@@ -36,6 +36,8 @@ def before_first_run(self, device, path, *args, **kwargs):
 
     def before_run(self, device, path, run, *args, **kwargs):
         super(WebExperiment, self).before_run(device, path, run)
+        device.shell('logcat -c')
```

Comment:
It seems to me that you should also add it to `NativeExperiment.py`. Or maybe even to `Experiment. before_run()` to make sure it's executed for other classes inheriting from `Experiment`.

Reply:
Hi, unfortunately I could not replicate the problems that you were having. Have you made sure that after your changes all of the cases still work? (Web, Native, Batterystat-web and Batterystat-Native) If so, let me know and we will merge your changes. 
Cheers

Reply:
@EMGrua Is this a question to me or in general for this pull request?

Reply:
@pawelulita sounds reasonable to do it from Experiment.

@EMGrua the adb calls are async, see https://github.com/ardevd/pyand/blob/master/pyand/ADB.py#L94, thus if your device is slow enough, your app will not have finished starting when the logs are cleared.

Reply:
@EMGrua I also experienced the problem with clearing logs, and moving it according to what @sktt suggested actually helped me solve this problem.